app/controllers/registers_controller.rb

```diff
@@ -63,12 +62,7 @@ def recover_records(fields, params, page_size = 100)
     if sort_by.present? && sort_direction.present?
       query = query.order("data->> '#{sort_by}' #{sort_direction.upcase} nulls last")
     end
-    @result_count = query.count
-    @offset = page_size * (page - 1) + 1
-    @offset_end = [@result_count, page_size * page].min
 
-    @total_pages = (@result_count / 100) + ((@result_count % 100).zero? ? 0 : 1)
-
-    query.page(page).per(page_size).without_count
+    query.page(params[:page]).per(100)
```

Comment:
This probably isn't a regression on what we were doing before, but do I understand correctly that we'll run the `where` statement on the query first, get back everything that matches that filter, and then only take the `params[:page]` page of 100 results?

Reply:
I think so, yes. We have stories in the icebox to refactor the two parts of this method anyway. Search and Sorting.